library/Vanilla/Models/SSOInfo.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Models;
+
+use Vanilla\ArrayAccessTrait;
+
+/**
+ * Class SSOInfo
+ *
+ * This is a data object containing information returned by an SSOAuthenticator.
+ * Adding methods to this object probably means that you are doing something wrong.
+ *
+ * @package Vanilla\Models
+ */
+class SSOInfo implements \ArrayAccess {
```

Comment:
This is probably me nitpicking, but the name of this class bugs me. I believe "data" is more commonly used in Vanilla. References to this class (and even it's description) refer to its contents as "data" too. I recommend either renaming it to "SSOData" or possibly even renaming it to "Data" and putting it under the "Vanilla\Models\SSO" namespace. The latter is a similar practice to what we see done with `Garden\Web\Data`.

Reply:
I can do SSOData.

Reply:
Done